src/list/whole/component.rs

```diff
@@ -186,6 +193,10 @@ where
             sort_list_kind: Rc::new(RefCell::new(None)),
 
             phantom: PhantomData,
+            view_modal: false,
+            modal_index: 0,
```

Comment:
Could we combine these two into one field of `Option<usize>`? I think `None` can represent no modal. (My suggestion doesn't cover the case where the modal index should be informed without modals.)

Reply:
Done